samplePDF/samplePDFFDBase.cpp

```diff
@@ -207,120 +332,55 @@ bool samplePDFFDBase::IsEventSelected(const std::vector< std::string >& Paramete
-{
-  MCSamples[sample].Oscillator->SetMNS(*oscpars[0], *oscpars[2], *oscpars[1], *oscpars[3], *oscpars[4], *oscpars[5], en, doubled_angle, nutype);
-  MCSamples[sample].Oscillator->propagateLinear(nutype , *oscpars[6], *oscpars[7]);
-
-  return MCSamples[sample].Oscillator->GetProb(nutype, oscnutype);
-}
-#endif
-
-//CalcOsc for Prob3++ GPU (ProbGpu)
-#if defined (USE_PROB3) && not defined (CPU_ONLY)
-extern "C" void setMNS(double x12, double x13, double x23, double m21, double m23, double Delta, bool kSquared);
-extern "C" void GetProb(int Alpha, int Beta, double Path, double Density, double *Energy, int n, double *oscw); 
-
-void samplePDFFDBase::calcOscWeights(int nutype, int oscnutype, double *en, double *w, int num)
-{
-  setMNS(*oscpars[0], *oscpars[2], *oscpars[1], *oscpars[3], *oscpars[4], *oscpars[5], doubled_angle);
-  GetProb(nutype, oscnutype, *oscpar[6], *oscpar[7], en, num, w);
-
-  if (std::isnan(w[10]))
-  {
-    MACH3LOG_ERROR("WARNING: ProbGPU oscillation weight returned NaN! {}", w[10]);
-  }
-}
-#endif
-
-//CalcOsc for CUDAProb3 CPU/GPU
-#if not defined (USE_PROB3)
-void samplePDFFDBase::calcOscWeights(int sample, int nutype, double *w)
-{
-  MCSamples[sample].Oscillator->setMNSMatrix(asin(sqrt(*oscpars[0])),asin(sqrt(*oscpars[2])), asin(sqrt(*oscpars[1])), (*oscpars[5]), nutype);
-  MCSamples[sample].Oscillator->setNeutrinoMasses(*oscpars[3], *oscpars[4]);
-  MCSamples[sample].Oscillator->calculateProbabilities(MCSamples[sample].NeutrinoType);
-  MCSamples[sample].Oscillator->getProbabilityArr(w, MCSamples[sample].ProbType);
-}
-#endif 
-
 void samplePDFFDBase::reweight() // Reweight function - Depending on Osc Calculator this function uses different CalcOsc functions
 {
-
-  if (Osc!=NULL) {
-	std::cout << "Osc is not NULL!! i.e. doing atm oscillations " << std::endl;
-    //DB Currently hardcoded to assume rho_electrons = rho_matter/2, 25km production height
-    Osc->FillOscillogram(oscpars,25.0,0.5);
-    for (unsigned int iSample=0;iSample<MCSamples.size();iSample++) {
-      for (int iEvent=0;iEvent<MCSamples[iSample].nEvents;iEvent++) {
-		MCSamples[iSample].osc_w[iEvent] = *(MCSamples[iSample].osc_w_pointer[iEvent]);
-      }
-    }
-  } else {
-    for (int i=0; i< (int)MCSamples.size(); ++i) {
-      
-#if defined (USE_PROB3) && defined (CPU_ONLY)
-      //Prob3 CPU needs to loop through events too
-      for(int j = 0; j < MCSamples[i].nEvents; ++j) {
-		MCSamples[i].osc_w[j] = calcOscWeights(i, MCSamples[i].nutype, MCSamples[i].oscnutype, *(MCSamples[i].rw_etru[j]));
-      } //event loop
-#endif
-      
-#if defined (USE_PROB3) && not defined (CPU_ONLY)
-      calcOscWeights(MCSamples[i].nutype, MCSamples[i].oscnutype, *(MCSamples[i].rw_etru), MCSamples[i].osc_w, MCSamples[i].nEvents);
-#endif
-      
-#if not defined (USE_PROB3)
-      calcOscWeights(i, MCSamples[i].nutype, MCSamples[i].osc_w);
-#endif
-    }// Sample loop
-  }
-
   //KS: Reset the histograms before reweight 
```

Comment:
thats what the function name says, we  shouldn't add noisey comments that write out function names longform, we should comment why that operation needs to be done. Why do we need to reset them? what does resetting them mean/do? set the bin contents to 0? nominal? do I need to know to understand the following code?